File: ruby/result.rb
#!/usr/bin/env ruby

require 'net/https'
require 'uri'
require './conf.rb'
if ARGV.size != 1
	puts 'Usage:'
	puts ' '+__FILE__+' delivery-id'
	exit
end
did = ARGV[0]

uri = URI.parse('https://api.bluehouselab.com/smscenter/v1.0/result/'+did)

http = Net::HTTP.new(uri.host, uri.port)
http.use_ssl = true
req = Net::HTTP::Get.new(uri.request_uri)
req.basic_auth Conf::APPID, Conf::APIKEY

resp = http.request(req)
puts resp.code
puts resp.body

File: php/result.php
<?php
include "conf.php";

if ($argc != 2) {
	error_log("Usage:\n  ".$argv[0]." delivery-id");
	exit(1);
}
$delivery_id = $argv[1];

$ch = curl_init("https://api.bluehouselab.com/smscenter/v1.0/result/".$delivery_id);
curl_setopt($ch, CURLOPT_TIMEOUT, 30);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
curl_setopt($ch, CURLOPT_USERPWD, "$APPID:$APIKEY");

$response = curl_exec($ch);
$response_header = curl_getinfo($ch);
echo $response_header["http_code"]."\n";
print_r(json_decode($response, true));
curl_close($ch);
?>
